Sources/NIOHTTPCompression/HTTPResponseCompressor.swift

```diff
@@ -66,6 +73,18 @@ public final class HTTPResponseCompressor: ChannelDuplexHandler, RemovableChanne
     public typealias OutboundIn = HTTPServerResponsePart
     /// This class emits `HTTPServerResponsePart` outbound.
     public typealias OutboundOut = HTTPServerResponsePart
+    
+    /// A closure that accepts a response header, optionally modifies it, and returns `true` if the response it belongs to should be compressed.
+    ///
+    /// - Parameter responseHeaders: The headers that will be used for the response. These can be modified as needed at this stage, to clean up any marker headers used to statelessly determine if compression should occur, and the new headers will be used when writing the response. Compression headers are not yet provided and should not be set; ``HTTPResponseCompressor`` will set them accordingly based on the result of this predicate.
+    /// - Parameter isCompressionSupported: Set to `true` if the client requested compatible compression, and if the HTTP response supports it, otherwise `false`.
+    /// - Returns: Return `true` if the compressor should proceed to compress the response, or `false` if the response should not be compressed.
+    ///
+    /// - Note: Returning `true` when compression is not supported will not enable compression, and the modified headers will always be used.
```

Comment:
This behaviour is surprising to me. I think we should introduce a new enum for the return value, something like `compressIfPossible` and `doNotCompress` as the cases. Just to clarify what will happen.

Reply:
I was thinking about this too, glad we are on the same page. Are we ok with `enum` enums, or should I make a struct with private enums for future compatibility?

Reply:
Went with a struct for futureproofness after all, but let me know if the cases are clearly named (went with an "intent"-like terminology, and clarified the wording in the doc comment)